src/coreclr/vm/threadsuspend.cpp

```diff
@@ -2509,6 +2509,18 @@ void RedirectedThreadFrame::ExceptionUnwind()
 #ifndef TARGET_UNIX
 
 #ifdef TARGET_X86
+
+// a helper to make sure these calls are not inlined into a filter function.
+// in particular we do not want a possibility of inlined destructor calls.
```

Comment:
This should not be a concern once you change it to `EEPOLICY_HANDLE_FATAL_ERROR`. `EEPOLICY_HANDLE_FATAL_ERROR` is non-inlineable.